launcher/firstLaunch/firstlaunch_moc.cpp

```diff
@@ -279,7 +293,69 @@ QString FirstLaunchView::getHeroesInstallDir()
 	return QString{};
 }
 
-void FirstLaunchView::copyHeroesData(const QString & path)
+void FirstLaunchView::extractGogData()
+{
+#ifndef VCMI_NO_INNOEXTRACT
+	QString filterExe = tr("GOG executable") + " (*.exe)";
+	QString fileExe = QFileDialog::getOpenFileName(this, tr("Select a GOG installer (exe) file..."), QDir::homePath(), filterExe);
+	if(fileExe.isEmpty())
+		return;
+
+	QString filterBin = tr("GOG bin file") + " (*.bin)";
+	QString fileBin = QFileDialog::getOpenFileName(this, tr("Select a GOG data (bin) file..."), QFileInfo(fileExe).absolutePath(), filterBin);
+	if(fileBin.isEmpty())
+		return;
+
+	if(QFileInfo(fileExe).completeBaseName() + QString("-1") != QFileInfo(fileBin).completeBaseName())
+	{
+		QMessageBox::critical(this, tr("File name error!"), tr("The filename of exe and bin has to be compatible!"), QMessageBox::Ok, QMessageBox::Ok);
+		return;
+	}
+
+	QTimer::singleShot(100, this, [this, fileExe, fileBin](){ // background to make sure FileDialog is closed...
+		QTemporaryDir dir;
+		if(dir.isValid() && QFile::exists(fileExe) && QFile::exists(fileBin)) {
+			ui->pushButtonGogInstall->setText(tr("Installing... Please wait!"));
+			QPalette pal = ui->pushButtonGogInstall->palette();
+			pal.setColor(QPalette::Button, QColor(Qt::yellow));
+			ui->pushButtonGogInstall->setAutoFillBackground(true);
+			ui->pushButtonGogInstall->setPalette(pal);
+			ui->pushButtonGogInstall->update();
+			ui->pushButtonGogInstall->repaint();
+
+			QString tmpFileExe = dir.filePath(QFileInfo(fileExe).fileName());
+			QFile(fileExe).copy(tmpFileExe);
+			QFile(fileBin).copy(dir.filePath(QFileInfo(fileBin).fileName()));
+
+			::extract_options o;
+			o.extract = true;
+
+			// standard settings
+			o.gog_galaxy = true;
+			o.codepage = (uint32_t)0;
+			o.output_dir = dir.path().toStdString();
+			o.extract_temp = true;
+			o.extract_unknown = true;
+			o.filenames.set_expand(true);
+
+			o.preserve_file_times = true; // also correctly closes file -> without it: on Windows the files are not written completly
+
+			process_file(tmpFileExe.toStdString(), o);
+
+			QStringList dirData = QDir(dir.path()).entryList({"data"}, QDir::Filter::Dirs);
+			if(dirData.empty() || QDir(QDir(dir.path()).filePath(dirData.front())).entryList({"*.lod"}, QDir::Filter::Files).empty())
+			{
+				QMessageBox::critical(this, tr("No Heroes III data!"), tr("Selected files does not contain Heroes III data!"), QMessageBox::Ok, QMessageBox::Ok);
+				ui->pushButtonGogInstall->setText(tr("Install GOG files"));
+				return;
+			}
+			copyHeroesData(dir.path(), true);
```

Comment:
We might need to clean up afterwards - delete all extracted files other than data/maps/mp3

Reply:
Not needed. Temp folder cleans up on destruction of object: https://doc.qt.io/qt-6/qtemporarydir.html

Reply:
Potential problem is that this temporary directory might be created on a different partition. Meaning:
- free space might be limited
- we won't be able to quickly move files from temporary directory to data directory if it is on different partition

Or it might not be a problem. Other than Linux where /tmp and /home are kind of recommended to be placed on different partitions, on rest of OS's our data directory is usually on same partition as OS and temp directories.

Reply:
Yes. Maybe not the perfect solution. But even under Linux it's the usual way to have the temp folder under the main partition. At least for the standard settings of many distributions-installer.

But I would leave it as it is. And maybe change later.

Reply:
While temp folder aka `/tmp` is usually on main partition, `/home` including /home/me/.local/share/vcmi (which is where we save h3 data) may be on a different partition.

But yeah, let's keep as it for now. Unlikely to be an issue. 1 Gb of temp files is low amount nowadays.